### PR TITLE
Pathlib vs OS Modules for File and Directory System Operation utilizing Comparable Scenarios

### LINKED ISSUE #5644: (Python) Pathlib vs OS Modules for File and Directory System Operation utilizing Comparable Scenarios
## Pathlib vs OS Modules for File and Directory System Operation utilizing Comparable Scenarios

In this article, we will run through different viable contexts to discover an agreeable decision of utilizing either Pathlib or OS for file system operation. Also, I will utilize this article to perform distinguished tasks for significant reference these tasks are not restricted to writing and reading a file, describing a document information, renaming files, creating directories and so on

The thought behind this article is to clear the examining choice knowing where every one of them fit in when working with File System Operation neither censuring a specific module, additionally to develop a quick consulting reference that is based on File System Operation.

### Key Takeaway
- It will help beginners in python or data analytics to confidently propose the right file system operation module.
- It will provide a simple road map for a quick guide at whatever point individuals intend to work with files.
- It will theoretically enhance beginners' comprehension of file system operation.
- It will eliminate the manual way of working with file system operations for individuals while working on a project.
- It will teach beginners how to reference an object in python, and this is pattern is called composition.

### Conclusion
At the end of this article I believe individuals would have understood the difference between Pathlib and OS, also I believe individuals would have been able to know where each module fit in and individuals would have been more familiar with File System Operation using two different types of File System Operation module.

Reference
[OS documentation](https://docs.python.org/3/library/os.html#querying-the-size-of-a-terminal)
[Pathib documentation](https://docs.python.org//pathlib.html)

### PR DESCRIPTION
**Pull request by akenz1901**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [ ] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [ ] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [ ] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [ ] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [ ] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [ ] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [ ] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [ ] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [ ] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [ ] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [ ] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [ ] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [ ] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [ ] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5644 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by srishilesh** [comment]
Hello @akenz1901 ,

Welcome to EngEd community!

Your PR has entered our reviewing queue. Please wait until a peer-reviewer pings you for revisions to be made.

Meanwhile, we suggest you to go through the published articles or approved PRs to see how the other authors have formatted their articles to avoid any potential closure of PRs.

Thanks.

**Comment by akenz1901** [comment]
@srishilesh Okay sir
Thanks i will be expecting the review report.

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 1]
```diff
@@ -0,0 +1,919 @@
...
+- [Retrieving and converting a file information](#retrieving-and-converting-a-file-information)
+- [Querying a file extension](#querying-a-file-extension)
+- [Wrapping Up](#wrapping-up)
+
+### Introduction
+File System Operation can be characterized as the example or technique that can be
+used to control or construct files and directories and there are two different ways
+of working with File System Operation, either utilizing functional dependency or utilizing
+Object-Oriented. I will practically highlight the different situations to separate between
+the two kinds of modules python provided for system operation. These modules are OS and Pathlib.
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:29]: Bring this to the top

Comment (srishilesh) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:29]: Avoid using "I"s. Make it third person.

Comment (srishilesh) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:29]: The whole paragraph has every sentence in a new line. Combine them together. 

Do this throughout the article

Comment (srishilesh) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:29]: Have a look at introductions in other articles to get an idea

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 2]
```diff
@@ -0,0 +1,919 @@
...
+
+#### OS Module
+The OS (operating system) module is a different operating system interface that functionally renders a convenient way
+of working with files and directories with a simple function call, additionally,
+OS empowers us to deal with I/O tasks straightforwardly.
+#### Pathlib Module
+The pathlib module is an Object-Oriented document framework activity that gives an order of just
+one direct superclass "PurePath" which permits you to play out a computational activity yet, doesn't
+give I/O activities, yet a substantial classpath that broadens PurePath and gives procedure on I/O.
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:38]: Please be more brief about PurePath.

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md (hunk 1 above)
Comment (srishilesh) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:29]: State why are we learning this? What are the objectives?

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 3]
```diff
@@ -0,0 +1,919 @@
...
+
+### Theoretical difference between Pathlib and Os
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:40]: You can move this section at the last. The reader must first gain an insight about the concept before trying to understand the difference

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 4]
```diff
@@ -0,0 +1,919 @@
...
+  of the class path for listing directory content which requires additional operation because method iterator returns an
+  object representation.
+  - Class path in pathlib provides a method for creating a permanent empty file with different extension which makes
+    working with File System Operation mind-boggling in python.
+- It can't be utilized to execute a document yet can open a record while composing or perusing a .txt document.
+- It doesn't make any arrangement for environment variables operation.
+- None of the available objects in pathlib makes a provision for making a duplicate file.
+
+### Practical difference between OS and Pathlib module
+In this section i will be illustrating the semantic differences between Pathlib and Os module, with comparable code
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:78]: Too many usage of "I" and inconsistent capitalization of letters

Comment (srishilesh) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:78]: Move this to the last

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 5]
```diff
@@ -0,0 +1,919 @@
...
+current path using os and pathlib will return an absolute path.
+
+Querying current path using os.
+```python
+import os
+
+os.getcwd()
+```
+Output:
+```output
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:96]: bash

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 6]
```diff
@@ -0,0 +1,919 @@
...
+Querying the current working directory is a basic task when you are dealing with a path.  There are two unique types of
+path which are relative path and absolute path.An absolute path gives a full description of the current location you are
+working while a relative path refers to a location that is relative to the current working directory. Note querying the
+current path using os and pathlib will return an absolute path.
+
+Querying current path using os.
+```python
+import os
+
+os.getcwd()
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:93]: Are you trying this out with notebook? Use print statements

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 7]
```diff
@@ -0,0 +1,919 @@
...
+output will be different and contains forward slash (/) if you are not using Windows os.
+
+Querying current path using pathlib.
+```python
+from pathlib import Path
+
+Path().cwd()
+```
+Output:
+```output
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:109]: bash

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 8]
```diff
@@ -0,0 +1,919 @@
...
+C:\Users\DELL\Desktop\practical_folder
+```
+The output returns a string representation of the current path you are working, note your
+output will be different and contains forward slash (/) if you are not using Windows os.
+
+Querying current path using pathlib.
+```python
+from pathlib import Path
+
+Path().cwd()
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:106]: Use print statement

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 9]
```diff
@@ -0,0 +1,919 @@
...
+#### Using os module.
+```python
+from os import path
+
+path.expanduser('~')
+```
+Output:
+```python
+C:\Users\DELL
+```
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:141]: Explain what is being done

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @akenz1901 ,

On going through your PR, I found some major changes to be made. I'm highlighting a few for you here:
- Make the intro briefer. Talk about the objectives.
- Avoid usage of "I"s.
- Don't write every sentence in a new line
- There are too many punctuation and grammar mistakes
- Capitalization of letters is inconsistent
- Code snippets were not explained properly
- The markdown format for code snippets is incorrect
- Use print() statements or mention that you are using notebooks
- Perform a grammar check
- Make the hero image dimensions to 600x400

Overall, I would say that the formatting and flow must be improved a lot. I would highly suggest you to go through approved PRs/ published articles to see how other authors have done it.

Ping me when done.

Since it requires several changes, I'll be closing this PR for now. Closing the PR doesn't mean it will not be published. 

Thanks.

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md (hunk 1 above)
Comment (akenz1901) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:29]: Okay @srishilesh 
Thank you so much for these illustrative corrections I will follow every step rendered above.

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md (hunk 2 above)
Comment (akenz1901) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:38]: Okay

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md (hunk 3 above)
Comment (akenz1901) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:40]: Okay

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md (hunk 5 above)
Comment (akenz1901) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:96]: Okay, thank you

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md (hunk 6 above)
Comment (akenz1901) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:93]: Yes sir
Okay
I will use print instead, but there's a reason I use notebook for some specific code snippets.

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md (hunk 7 above)
Comment (akenz1901) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:109]: Okay, thank you

**Comment by akenz1901** [comment]
Hi, @srishilesh I have just reuploaded it and I have corrected it as expected.
Thank you so much for the guide.

**Comment by srishilesh** [comment]
> Hi, @srishilesh I have just reuploaded it and I have corrected it as expected.
> Thank you so much for the guide.

Hey @akenz1901 ,

Thanks for creating a new PR.

Can you ping me in the PR too? Mention that it's a duplicate of this PR.

I'll check on it later (by tomorrow)

Thanks.

**Comment by akenz1901** [comment]
Hi @srishilesh 
Duplicate of #6145

**Pull request closed without merging**